src/Filters/buttord.jl

```diff
@@ -0,0 +1,187 @@
+    returns a non-integer BSF order estimate.
+    """
+
+    # override one of the passband edges with the test frequency.
+    Wpc = uselowband ? [Wx, Wp[2]] : [Wp[1], Wx]
+    
+    # get the new warp frequency.
+    warp = minimum(abs.((Ws .* (Wpc[1]-Wpc[2])) ./ (Ws.^2 .- (Wpc[1]*Wpc[2]))))
+
+    # use the new frequency to determine the new filter order.
+    order_estimate(Rp, Rs, warp)
+end
+
+toprototype(Wp::Real, Ws::Real, ftype::Type{Lowpass}) = Ws / Wp
+toprototype(Wp::Real, Ws::Real, ftype::Type{Highpass}) = Wp / Ws
+
+function toprototype(Wp::AbstractArray{<:Real}, Ws::AbstractArray{<:Real}, ftype::Type{Bandpass}) 
+    # bandpass filter must have two corner frequencies we're computing with
+    Wa = (Ws.^2 .- Wp[1]*Wp[2]) ./ (Ws .* (Wp[1]-Wp[2]))
+    minimum(abs.(Wa))
+end
+
+function toprototype!(Wp::AbstractArray{<:Real}, Ws::AbstractArray{<:Real}, Rp::Real, Rs::Real, ftype::Type{Bandstop})
+    # NOTE: the optimization function will adjust the corner frequencies in Wp, modifying the original input.
+    Δ = eps(typeof(Wp[1]))^(2/3)
+
+    # optimize order on bound [passband low < w < stopband-tolerance].
+    C₁(w) = bsfcost(w, true, Wp, Ws, Rp, Rs)
+    p1 = minimizer(optimize(C₁, Wp[1], Ws[1]-Δ))
+    Wp[1] = p1 # modifying band edge for next optimization run.
+    
+    # declaring the 2nd cost function here to make sure the overwritten 
+    # passband edge vector Wp is used in the new cost function.
+    C₂(w) = bsfcost(w, false, Wp, Ws, Rp, Rs)
+    p2 = minimizer(optimize(C₂, Ws[2]+Δ, Wp[2]))
+    Wp[2] = p2
+
+    Wa = (Ws .* (Wp[1]-Wp[2])) ./ (Ws.^2 .- (Wp[1]*Wp[2]))
+    minimum(abs.(Wa))
+end
+
+fromprototype(Wp::Real, Wscale::Real, ftype::Type{Lowpass}) = Wp * Wscale
+fromprototype(Wp::Real, Wscale::Real, ftype::Type{Highpass}) = Wp / Wscale
+
+function fromprototype(Wp::AbstractArray{<:Real}, Wscale::Real, ftype::Type{Bandstop})
+    Wa = zeros(2)
+    diff = Wp[2]-Wp[1]
+    prod = Wp[2]*Wp[1]
+    Wa[1] = (diff + sqrt(diff^2 + 4*(Wscale^2)*prod)) / (2*Wscale)
+    Wa[2] = (diff - sqrt(diff^2 + 4*(Wscale^2)*prod)) / (2*Wscale)
+    sort!(abs.(Wa))
+end
+
+function fromprototype(Wp::AbstractArray{<:Real}, Wscale::Real, ftype::Type{Bandpass})
+    Wsc = [-Wscale, Wscale]
+    Wa = -Wsc .* (Wp[2]-Wp[1])./2 .+ sqrt.( Wsc.^2/4*(Wp[2]-Wp[1]).^2 .+ Wp[1]*Wp[2])
+    sort!(abs.(Wa))
+end
+
+order_estimate(Rp::Real, Rs::Real, warp::Real) = (log10(^(10, 0.1*Rs) - 1) - log10(^(10, 0.1*Rp) - 1)) / (2*log10(warp))
+natfreq_estimate(warp::Real, Rs::Real, order::Integer) = warp / (^(10, 0.1*Rs) - 1)^(1/(2*order))
+
+function buttord(Wp::AbstractArray{<:Real}, Ws::AbstractArray{<:Real}, Rp::Real, Rs::Real)
+    """
+        buttord(Wp, Ws, Rp, Rs)
+
+    Butterworth order estimate for bandpass and bandstop filter types. 
+    `Wp` and `Ws` are 2-element pass/stopband frequency edges, with filttype specifying
+    the `Bandpass` or `Bandstop` filter types.
+    """
```

Comment:
Docstrings need to go before the function definition. Also, this should mention/explain the return value(s), which is not just the estimated order.

Reply:
Moved and updated the docstrings!